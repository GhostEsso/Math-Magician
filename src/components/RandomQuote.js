import React, { useState, useEffect } from 'react';

function RandomQuote() {
  const [quote, setQuote] = useState(null);
  const [author, setAuthor] = useState(null);
  const [error, setError] = useState(null);

  useEffect(() => {
    const dataFetch = async () => {
      try {
        const response = await fetch('https://api.api-ninjas.com/v1/quotes?category=happiness', {
          method: 'GET',
          headers: {
            'X-Api-Key': 'V8FzqJZj+pOipGZbfuVsdw==vKvb8VNKkoGE0fLQ',
            'Content-type': 'application/json',
          },
        });
        const quotes = await response.json();
        if (quotes.length > 0) {
          setQuote(quotes[0].quote);
          setAuthor(quotes[0].author); // Set the author of the quote
        } else {
          setQuote('');
          setAuthor('');
        }
      } catch (err) {
        setError(err.message);
      }
    };
    dataFetch();
  }, []);

  if (error) {
    return (
      <div className="Quotes">
        <p>
          Error:
          {error}
        </p>
      </div>
    );
  }

  return (
    <div className="Quotes">
      {quote ? (
        <div>
          <p>{quote}</p>
          {author && (
          <p>
            -
            {author}
          </p>
          )}
        </div>
      ) : (
        <p>Loading ...</p>
      )}
    </div>
  );
}

export default RandomQuote;
