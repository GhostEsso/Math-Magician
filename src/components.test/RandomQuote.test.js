import React from 'react';
import renderer from 'react-test-renderer';
import RandomQuote from '../components/RandomQuote';

test('Snapshot test for RandomQuote component', () => {
  const component = renderer.create(<RandomQuote />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
