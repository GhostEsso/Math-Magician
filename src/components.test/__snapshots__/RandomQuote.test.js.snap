// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test for RandomQuote component 1`] = `
<div
  className="Quotes"
>
  <p>
    Loading ...
  </p>
</div>
`;
